xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -732,7 +738,16 @@ class CVideoPlayActionProcessor : public CVideoPlayActionProcessorBase
     const ContentUtils::PlayMode mode{m_item.GetProperty("CheckAutoPlayNextItem").asBoolean()
                                           ? ContentUtils::PlayMode::CHECK_AUTO_PLAY_NEXT_ITEM
                                           : ContentUtils::PlayMode::PLAY_ONLY_THIS};
-    VIDEO_UTILS::PlayItem(std::make_shared<CFileItem>(m_item), "", mode);
+
+    if (m_item.GetVideoInfoTag()->m_type == MediaTypeMovie)
+    {
+      CGUIDialogVideoVersion::PlayVideoVersion(std::make_shared<CFileItem>(m_item),
+                                               [mode](const std::shared_ptr<CFileItem>& item) {
+                                                 VIDEO_UTILS::PlayItem(item, "", mode);
+                                               });
+    }
+    else
+      VIDEO_UTILS::PlayItem(std::make_shared<CFileItem>(m_item), "", mode);
```

Comment:
After our previous discussions on this one, I thought again for this problem and believed keeping the current way may be a better solution. The major problem here is, each place playing an item has special processing for the item (updating some properties of the item, etc.), if we put this in a central place (base class), we will lose those special processing. I'd like to have a better central place solution if we can solve the above problem.

Reply:
Yes, this is all a bit messy still in current code base, and I don't have a solution at hand, was just offering to think about it and in case I have something to share it.